semantic-search/src/test/java/org/examples/OpenAIHandlerTest.java

```diff
@@ -0,0 +1,115 @@
+package org.examples;
+
+import com.theokanning.openai.embedding.Embedding;
+import com.theokanning.openai.embedding.EmbeddingResult;
+import com.theokanning.openai.model.Model;
+import com.theokanning.openai.service.OpenAiService;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class OpenAIHandlerTest {
+
+    @Test
+    public void testListModelsContainsExpectedNumberAndContents() {
+        OpenAiService mockService = mock(OpenAiService.class);
```

Comment:
**very nit:** Could technically share the `mockService` across tests and use a `@BeforeEach` to reset the mock for each test for setup.

Reply:
Oo nice, I'll do that thanks!